src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -786,63 +862,53 @@ private static QuickAttributes GetQuickAttributes(SyntaxList<AttributeListSyntax
             return result;
         }
 
-        private static readonly ObjectPool<ImmutableSegmentedHashSet<string>.Builder> s_memberNameBuilderPool =
-            new ObjectPool<ImmutableSegmentedHashSet<string>.Builder>(() => ImmutableSegmentedHashSet.CreateBuilder<string>());
-
-        private static ImmutableSegmentedHashSet<string> ToImmutableAndFree(ImmutableSegmentedHashSet<string>.Builder builder)
-        {
-            var result = builder.ToImmutable();
-            builder.Clear();
-            s_memberNameBuilderPool.Free(builder);
-            return result;
-        }
-
-        private static ImmutableSegmentedHashSet<string> GetEnumMemberNames(SeparatedSyntaxList<EnumMemberDeclarationSyntax> members, ref SingleTypeDeclaration.TypeDeclarationFlags declFlags)
+        private BoxedMemberNames GetEnumMemberNames(
+            EnumDeclarationSyntax enumDeclaration,
+            SeparatedSyntaxList<EnumMemberDeclarationSyntax> members,
```

Comment:
Would it be simpler if we just had a parameter for the `Internal.EnumDeclarationSyntax` and accessed its Members property in here? Similar for other `GetNonTypeMemberNames` methods.

Reply:
Can do for the enum case.  For the non-type case it's not feasible as we pass different parents (like a CompilationUnit) and dont' have a common way to get at the child members.